nmdc_automation/re_iding/db_utils.py

```diff
@@ -36,6 +36,69 @@
     NOM_ANALYSIS_ACTIVITY_SET
 ]
 
+WORKFLOW_ID_CODE_MAP = {
+    "nmdc:wfmag": MAGS_ACTIVITY_SET,
+    "nmdc:wfmb": METABOLOMICS_ANALYSIS_ACTIVITY_SET,
+    "nmdc:wfmgan": METAGENOME_ANNOTATION_ACTIVITY_SET,
+    "nmdc:wfmgas": METAGENOME_ASSEMBLY_SET,
+    "nmdc:wfrbt": READ_BASED_TAXONOMY_ANALYSIS_ACTIVITY_SET,
+    "nmdc:wfrqc": READS_QC_SET,
+}
+
+
+def get_collection_name_from_workflow_id(workflow_id: str) -> str:
+    """
+    Get the collection name for the given workflow ID based on the
+    embedded code.
+     e.g. nmdc:wfrqc-11-zbyqeq59.1 -> nmdc:wfrqc -> READS_QC_SET
+    """
+    code = workflow_id.split("-")[0]
+    return WORKFLOW_ID_CODE_MAP.get(code, "unknown")
+
+
+def fix_malformed_workflow_id_version(workflow_id: str) -> str:
+    """
+    Fix a malformed workflow id version examples
+        - extra .1(s) at the end of the version e.g. nmdc:wfrqc-11-zbyqeq59.1.1
+        - missing version e.g nmdc:wfrqc-11-zbyqeq59
+    Corrected version will be nmdc:wfrqc-11-zbyqeq59.1
+    """
+    parts = workflow_id.split(".")
```

Comment:
it would be more robust here to use the materialized patterns derived from the schema,
that is
^(nmdc):(wfmag|wfmb|wfmgan|wfmgas|wfmsa|wfmp|wfmt|wfmtan|wfmtas|wfnom|wfrbt|wfrqc)-([0-9][a-z]{0,6}[0-9])-([A-Za-z0-9]{1,})(\\.[0-9]{1,})$